### PR TITLE
Feature/database integration with jdbc

### PR DESCRIPTION
**Pull request by 8r1m2t0n3**

### CODE REVIEW

File: src/main/java/org/andersen/homework/model/dao/jdbc/TicketDaoJdbc.java [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+    return ticket;
+  }
+
+  @Override
+  public void update(UUID id, Ticket ticket) {
+    try (PreparedStatement preparedStatement = connection.prepareStatement(UPDATE_QUERY)) {
+      preparedStatement.setString(1, ticket.getType().name());
+      preparedStatement.setBigDecimal(2, ticket.getPriceInUsd());
+
+      if (ticket instanceof BusTicket busTicket) {
```
Comment (Darya-Drutsko) [inline src/main/java/org/andersen/homework/model/dao/jdbc/TicketDaoJdbc.java:76]: It's kind of duplicated code, so separate please in method and reuse it

File: src/main/java/org/andersen/homework/model/dao/jdbc/TicketDaoJdbc.java [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+      preparedStatement.setObject(1, id);
+
+      preparedStatement.execute();
+    } catch (SQLException e) {
+      throw new TicketDeletingErrorException(e);
+    }
+  }
+
+  @Override
+  public Ticket get(UUID id) {
```
Comment (Darya-Drutsko) [inline src/main/java/org/andersen/homework/model/dao/jdbc/TicketDaoJdbc.java]: rename to getById

Comment (Darya-Drutsko) [reply src/main/java/org/andersen/homework/model/dao/jdbc/TicketDaoJdbc.java]: And what about getting ticket by user ID ?

File: src/main/resources/db/sql/ticket_and_user_tables_creation.sql [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    ticket_class VARCHAR(50), -- Specific to BusTicket
+    duration VARCHAR(50), -- Specific to BusTicket
+    start_date DATE, -- Specific to BusTicket
+    CHECK (type IN ('CONCERT', 'BUS'))
+);
+
+CREATE TABLE IF NOT EXISTS "user" (
+    id UUID PRIMARY KEY,
+    role VARCHAR(50) NOT NULL,
+    ticket_id UUID UNIQUE REFERENCES ticket(id), -- Specific to Client
```
Comment (Darya-Drutsko) [inline src/main/resources/db/sql/ticket_and_user_tables_creation.sql]: Why in user table we have ticket_id? 
What type of relation should be here?

File: src/main/java/org/andersen/homework/model/dao/jdbc/UserDaoJdbc.java [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+      }
+    } catch (SQLException e) {
+      throw new UserUpdatingErrorException(e);
+    }
+  }
+
+  @Override
+  public void delete(UUID id) {
+    try (PreparedStatement preparedStatement = connection.prepareStatement(DELETE_QUERY)) {
+      preparedStatement.setObject(1, id);
```
Comment (Darya-Drutsko) [inline src/main/java/org/andersen/homework/model/dao/jdbc/UserDaoJdbc.java:123]: What about deleting tickets for related user?)

File: src/main/java/org/andersen/homework/util/TransactionalConnectionManager.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+      } catch (SQLException e) {
+        throw new RuntimeException(e);
+      }
+    }
+  }
+
+  public void releaseSavepoint() {
+    if (savepoint != null) {
+      try {
+        connection.releaseSavepoint(savepoint);
```
Comment (Darya-Drutsko) [inline src/main/java/org/andersen/homework/util/TransactionalConnectionManager.java:74]: What is the main purpose of releaseSavepoint method?

Comment (8r1m2t0n3) [reply src/main/java/org/andersen/homework/util/TransactionalConnectionManager.java:74]: The method removes last created savepoint, accessing creation of a new one

File: src/main/resources/db/sql/ticket_and_user_tables_creation.sql (hunk 3 above)
Comment (8r1m2t0n3) [reply src/main/resources/db/sql/ticket_and_user_tables_creation.sql]: Removed `ticket_id` field from `user` and added `user_id` to `ticket`, added cascade delete also. Now tickets remove after owning users deletion

**Pull request merged**